### PR TITLE
jfs - Isawa Heiko

### LINKED ISSUE #3379: JfS - Isawa Heiko
image

### PR DESCRIPTION
**Pull request by Matthew-Griffith**
closes #3379

### CODE REVIEW

File: server/game/cards/09.3-JfS/IsawaHeiko.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                        context.game.isDuringConflict() &&
+                        event.player === context.player;
+                }
+            },
+            target: {
+                cardType: CardTypes.Character,
+                cardCondition: card => !card.hasDash() && card.isParticipating(),
+                gameAction: AbilityDsl.actions.cardLastingEffect({
+                    effect: AbilityDsl.effects.switchBaseSkills()
+                })
```
Comment (jamesoddy) [inline server/game/cards/09.3-JfS/IsawaHeiko.js]: It looks as though Heiko effect is persistent without an expiration.  Please add `duration: Durations.persistent`

File: test/server/cards/09.3-JfS/IsawaHeiko.spec.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                this.player2.pass();
+                this.player1.clickCard(this.against);
+                this.player1.clickCard(this.solemn);
+                this.player1.clickCard(this.heiko);
+                this.player1.clickCard(this.makoto);
+                expect(this.makoto.getMilitarySkill()).toBe(1);
+                expect(this.makoto.getPoliticalSkill()).toBe(4);
+            });
+
+            it('should target characters at home', function() {
```
Comment (jamesoddy) [inline test/server/cards/09.3-JfS/IsawaHeiko.spec.js]: I think you mean `should *not* target characters at home`

File: server/game/cards/09.3-JfS/IsawaHeiko.js (hunk 1 above)
Comment (naesseth) [reply server/game/cards/09.3-JfS/IsawaHeiko.js]: The actual card has expiration (end of phase), just a miss by 5rdb.

Comment (jamesoddy) [reply server/game/cards/09.3-JfS/IsawaHeiko.js]: Yep, just been informed.  Either way this still needs a change, this now needs to be `duration: Durations.UntilEndOfPhase`

File: server/game/cards/09.3-JfS/IsawaHeiko.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            when: {
+                onCardPlayed: (event, context) => {
+                    return event.card.hasTrait(Elements.Water) &&
+                        context.game.isDuringConflict() &&
+                        event.player === context.player;
+                }
+            },
+            target: {
+                cardType: CardTypes.Character,
+                cardCondition: card => !card.hasDash() && card.isParticipating(),
```
[the remark this answers is not in the document]
Comment (Matthew-Griffith) [reply server/game/cards/09.3-JfS/IsawaHeiko.js]: yeah without that heiko can target the goblin sneak in the tests

[the remark this answers is not in the document]
Comment (jamesoddy) [reply server/game/cards/09.3-JfS/IsawaHeiko.js]: Ok, we should probably take a look at that, but no harm this being here anyway.

**Pull request merged**